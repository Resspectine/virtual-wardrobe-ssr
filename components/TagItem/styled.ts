import { Cancel } from '@mui/icons-material';
import { Box, styled, Typography } from '@mui/material';

export const TagItemWrapper = styled(Box)(({ theme }) => ({
  maxHeight: 40,
  padding: theme.spacing(0.625, 2),
  backgroundColor: '#f5f5f5',
  borderRadius: '100vh',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  cursor: 'default',
  margin: theme.spacing(0, 0.625),
  marginBottom: theme.spacing(1.25),
}));

export const TagItemTitle = styled(Typography)({
  fontSize: 14,
  lineHeight: '14px',
});

export const TagCancel = styled(Cancel)(({ theme }) => ({
  fontSize: 20,
  color: theme.palette.text.secondary,
  marginLeft: 5,
  cursor: 'pointer',
  '&:hover': {
    color: '#595959',
  },
}));
