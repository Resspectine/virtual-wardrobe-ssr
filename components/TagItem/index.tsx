import Box from '@mui/material/Box';
import { FC, MouseEventHandler } from 'react';

import { TagCancel, TagItemTitle, TagItemWrapper } from './styled';

import { Tag } from '@/types/tag';

interface TagsItemProps {
  tag: Tag;
  onDelete?: (id: string) => MouseEventHandler<HTMLDivElement>;
}

const TagItem: FC<TagsItemProps> = ({ tag, onDelete }) => (
  <TagItemWrapper>
    <TagItemTitle>{tag.title}</TagItemTitle>
    {onDelete && (
      <Box display="flex" onClick={onDelete(tag.id)}>
        <TagCancel />
      </Box>
    )}
  </TagItemWrapper>
);

export default TagItem;
