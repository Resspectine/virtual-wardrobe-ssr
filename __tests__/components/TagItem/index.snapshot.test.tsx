import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import TagItem from '@/components/TagItem';

describe('TagItem', () => {
  test('should display tag title', () => {
    const { asFragment } = render(
      <TagItem
        tag={{
          id: 'id',
          title: 'Title',
        }}
      />
    );

    expect(asFragment).toMatchSnapshot();
  });

  test('should display delete button', () => {
    const { asFragment } = render(
      <TagItem
        tag={{
          id: 'id',
          title: 'Title',
        }}
        onDelete={() => () => {}}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
