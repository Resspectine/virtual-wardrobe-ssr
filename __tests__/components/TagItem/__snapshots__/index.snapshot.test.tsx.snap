// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagItem should display delete button 1`] = `
<DocumentFragment>
  .emotion-0 {
  max-height: 40px;
  padding: 5px 16px;
  background-color: #f5f5f5;
  border-radius: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: default;
  margin: 0px 5px;
  margin-bottom: 10px;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  font-size: 14px;
  line-height: 14px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  font-size: 20px;
  color: rgba(0, 0, 0, 0.6);
  margin-left: 5px;
  cursor: pointer;
}

.emotion-3:hover {
  color: #595959;
}

<div
    class="MuiBox-root emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 emotion-1"
    >
      Title
    </p>
    <div
      class="MuiBox-root emotion-2"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-3"
        data-testid="CancelIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TagItem should display tag title 1`] = `[Function]`;
